### DISCUSSION TITLE
How to work around Identity.Web during Integration Tests

### DISCUSSION THREAD

**Discussion by IdpugantiSanjay** [AzureAD/microsoft-identity-web#1159] [Q&A]
The question says it all. How to work around the `Authorize` attribute and `HttpContext.VerifyUserHasAnyAcceptedScope` during Integration Tests

**Comment by jmprieur** [comment]
@IdpugantiSanjay : you cannot really. you need to provide a token.
Maybe our integration tests (for a web api) can help you?:
https://github.com/AzureAD/microsoft-identity-web/blob/master/tests/Microsoft.Identity.Web.Test.Integration/AcquireTokenForUserIntegrationTests.cs

**Reply by jennyf19** [reply]
@IdpugantiSanjay do our integration tests help at all? if not, what exactly are you looking to do?

**Reply by scrocquesel-ml150** [reply]
@jmprieur Could it be more easier and have something that can be closer to unit tests without network call ?
